### PR TITLE
Skip tests requiring fetched data

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description

This test requires a dataset that isn't available.

But we need to skip it.

I want to make sure there aren't any other ones that we need to skip.
xref: https://github.com/conda-forge/scikit-image-feedstock/pull/85

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/data/tests/test_data.py [hunk 1]
```diff
@@ -149,6 +149,9 @@ def test_skin():
 
     Needs internet connection.
     """
+    # Fetch causes the test to get skipped if it isn't available during
```
Comment (mkcor) [inline skimage/data/tests/test_data.py:152]:
```suggestion
    # Fetch causes the test to get skipped if pooch isn't available during
```
?

**Review by mkcor — APPROVED** [review]
Is the PR actually WIP?

**Comment by hmaarrfk** [comment]
Yes, it is actually a WIP

**Comment by hmaarrfk** [comment]
@mkcor I actually just forced pushed something that I feel is more maintainable. I will likely want to remove the other testing util. I'm testing how this works now 
https://github.com/conda-forge/scikit-image-feedstock/pull/85

**Review by grlee77 — APPROVED** [review]
I like that this solution does not rely on having to use a special test utility.

I guess we still need to make sure to load data from within a test function rather than at global scope, though, as done here for `test_blur_effect.py`. Are there still possibly other files where we need to make a similar change?

**Comment by hmaarrfk** [comment]
There is one other location in the `rank` tests that try to avoid multiple threads fetching the data in parallel.

**Comment by hmaarrfk** [comment]
@grlee77 do you want me to to introduce the cleanup in here as well? as in, removing the custom utility function? I wanted to keep the conda-forge patch small.

**Comment by grlee77** [comment]
> @grlee77 do you want me to to introduce the cleanup in here as well? as in, removing the custom utility function? I wanted to keep the conda-forge patch small.

It is fine to have that in a follow-up if you prefer. Do you want to get these changes in prior to tagging 0.19.1? If so, please add the milestone tag here (I am hoping to make a new 0.19.1 release soon given that I removed broken arm64 wheels from PyPI for 0.19.0)

**Comment by hmaarrfk** [comment]
I do think this will help many packagers (who are not allowed to download during the test phase) move things forward and include scikit-image 0.19.1 in their repos.

**Comment by grlee77** [comment]
Sure. Please just remove the WIP from the title when this is ready for review.

**Comment by grlee77** [comment]
anything I can help with here @hmaarrfk? (this is the only remaining PR tagged for 0.19.1)

File: skimage/data/_fetchers.py [hunk 2]
```diff
@@ -209,6 +221,17 @@ def _fetch(data_filename):
...
+        # Packagers might use pytest to run the tests suite, but may not
+        # want to run it online.
+        # As such, we will avoid failing the test, and silently skipping it.
+        if 'PYTEST_CURRENT_TEST' in os.environ:
+            # https://docs.pytest.org/en/latest/example/simple.html#pytest-current-test-environment-variable
+            import pytest
+            # Pytest skip raises an exception that allows the
+            # tests to be skipped
+            pytest.skip(f'Unable to download {data_filename}',
+                        allow_module_level=True)
```
Comment (grlee77) [inline skimage/data/_fetchers.py:234]: since this looks duplicate with the above, perhaps refactor into a small, private helper function?

**Comment by grlee77** [comment]
confirmed locally that running the test suite without pooch passes (17 additional tests were skipped vs. when it is installed)

**Comment by grlee77** [comment]
> There is one other location in the rank tests that try to avoid multiple threads fetching the data in parallel.

Okay, I see that this one already skips when pooch is present, but will need to be updated if we remove the existing `fetch` utility. (`data/rank_filters_tests_3d.npz` is on the gitlab repo)

**Comment by grlee77** [comment]
I updated one of the GitHub Actions test runs to uninstall pooch prior to calling `pytest`.

**Comment by grlee77** [comment]
verified in log of the CI run that pooch got uninstalled as intended: https://github.com/scikit-image/scikit-image/runs/4463723438?check_suite_focus=true#step:5:235

**Comment by hmaarrfk** [comment]
Thanks!

**Comment by hmaarrfk** [comment]
I'm going to rebase the PR into two comments
one code related, the other CI related.

**Comment by hmaarrfk** [comment]
I simply think this will allow downstream to pick the patch tey need in a more straightforward manner since the CI stuff isn't included in the pip package.

**Review by grlee77 — APPROVED** [review]
Looks good to me. Thanks @hmaarrfk!

**Comment by grlee77** [comment]
@meeseeksdev backport to v0.19.x

**Pull request merged**